zerver/tests/test_auth_backends.py

```diff
@@ -110,17 +110,20 @@ def verify_backend(self, backend: Any, good_kwargs: Optional[Dict[str, Any]]=Non
 
         # Verify auth fails with a deactivated user
         do_deactivate_user(user_profile)
-        result = backend.authenticate(**good_kwargs)
-        if isinstance(backend, SocialAuthMixin):
-            # Returns a redirect to login page with an error.
-            self.assertEqual(result.status_code, 302)
-            self.assertEqual(result.url, "/login/?is_deactivated=true")
-        else:
-            # Just takes you back to the login page treating as
-            # invalid auth; this is correct because the form will
-            # provide the appropriate validation error for deactivated
-            # account.
-            self.assertIsNone(result)
+        with mock.patch('zproject.backends.logging.info') as m:
+            result = backend.authenticate(**good_kwargs)
+            if isinstance(backend, SocialAuthMixin):
+                # Returns a redirect to login page with an error.
+                self.assertEqual(result.status_code, 302)
+                self.assertEqual(result.url, "/login/?is_deactivated=true")
+                m.assert_called_with("Failed attempt to login from an inactive account with ID: %s",
```

Comment:
Should we really be checking the exact log line here? It's already done in `test_social_auth_deactivated_user`, so this seems a bit redundant.

Looks good to me other than that! :slightly_smiling_face:

Reply:
Oh, true. It seems a bit redundant. Remove that.